bridge-sdk/bridge-clients/solana-bridge-client/src/solana_bridge_client.rs

```diff
@@ -242,12 +237,14 @@ impl SolanaBridgeClient {
             ],
             &spl_associated_token_account::ID,
         );
-        let (vault, _) = Pubkey::find_program_address(&[b"vault", token.as_ref()], program_id);
 
         let (wormhole_bridge, wormhole_fee_collector, wormhole_sequence) =
             self.get_wormhole_accounts().await?;
         let wormhole_message = Keypair::new();
 
+        let token_owner = self.get_token_owner(token).await?;
+        let is_solana_native = token_owner.is_some() && token_owner.unwrap() == authority;
```

Comment:
Why do we use `unwrap` here? Let's remove it and parse `token_owner` in place. For example:
```rust
let Some(token_owner) = self.get_token_owner(token).await? else {
    return Err(SolanaClientError::ConfigError("Failed to get token owner"));
}
let is_solana_native = token_owner == authority;

Reply:
Fixed in [6818466](https://github.com/Near-One/bridge-sdk-rs/pull/60/commits/68184669fa88dec4905e0886a66cc5139aac4620)